proto/cusf/mainchain/v1/validator.proto

```diff
@@ -190,24 +173,23 @@ message GetSidechainProposalsRequest {
 }
 message GetSidechainProposalsResponse {
   message SidechainProposal {
-    uint32 sidechain_number = 1;
+    google.protobuf.UInt32Value sidechain_number = 1;
 
     // protolint:disable MAX_LINE_LENGTH
 
-    // Raw sidechain proposal data (M1 message of BIP300)
-    // https://github.com/bitcoin/bips/blob/master/bip-0300.mediawiki#m1----propose-sidechain
-    google.protobuf.BytesValue data = 2;
+    // Raw sidechain proposal description
+    cusf.common.v1.ConsensusHex description = 2;
 
     // protolint:enable MAX_LINE_LENGTH
 
     // Sidechain data, as declared in the M1 proposal.
     // Might be nil, if the proposal uses an unknown version.
     optional SidechainDeclaration declaration = 7;
 
-    cusf.common.v1.ConsensusHex data_hash = 3;
-    uint32 vote_count = 4;
-    uint32 proposal_height = 5;
-    uint32 proposal_age = 6;
+    cusf.common.v1.ReverseHex description_sha256d_hash = 3;
+    google.protobuf.UInt32Value vote_count = 4;
+    google.protobuf.UInt32Value proposal_height = 5;
+    google.protobuf.UInt32Value proposal_age = 6;
```

Comment:
Why are these wrappers? Same goes for line 176

Reply:
[Primitive values have **_implicit presence_** in proto3](https://protobuf.dev/programming-guides/field_presence/). If they are set to the default value, they are not included in a serialized message, and if they are not included in a serialized message, they are assumed to have default value.

This is usually not what is desired - the consumer cannot distinguish between a missing field and the default value, so if a proto is updated to deprecate a field, the consumer will receive a default value instead of knowing that the value was not included.

Explicit field presence can be enabled with `optional`, or with `google.protobuf.*Value` messages. I dislike the use of `optional` to simply enable explicit field presence - `optional` is best used when a field is actually optional. This motivates the following convention:

* Always use explicit field presence, unless the default value is always invalid and can be re-used to indicate field presence
* Use  `google.protobuf.*Value` messages to enable explicit field presence
* Use `optional` to indicate that a field is optional

ie.

```proto
message Example {
    // Empty string is always invalid for an address, so this is fine
    string address = 1;
    // Empty string is always invalid for an address,
    // so this is ok to indicate an optional address
    optional string other_address = 2;
    // Zero amounts are valid, use explicit field presence
    google.protobuf.UInt32Value amount = 3;
    // Zero amounts are valid, and the value is optional,
    // so we can distinguish missing vs unspecified vs default
    optional google.protobuf.UInt32Value other_amount = 4;
}
```

Reply:
I get your reasoning, and you have valid points. My opinion on this is that it introduces more complexity than is necessary. In my usage of Protobuf schemas over the years I've avoided using wrapper values _and_ optionals for the vast, vast majority of requests/responses without issue. I think using raw values would work nicely here, and save us some extra wrapper usage. 

I'll leave it up to you to decide!